src/linear_algebra/trace.lean

```diff
@@ -115,6 +116,28 @@ end
 section
 variables (R : Type u) [field R] {M : Type v} [add_comm_group M] [module R M]
 
+/-- The trace of a linear map correspond to the contraction pairing under the isomorphism
+ `M →ₗ M ≃ M* ⊗ M`-/
+lemma trace_eq_contract [finite_dimensional R M] :
+  (linear_map.trace R M) ∘ₗ ↑(dual_tensor_hom_equiv R M M) = contract_left R M :=
```

Comment:
```suggestion
  (linear_map.trace R M) ∘ₗ (dual_tensor_hom R M M) = contract_left R M :=
```
For this to be a `simp` lemma (and it should), it's LHS cannot contain anything that is `simp`able. If we add `coe_dual_tensor_hom_equiv_to_linear` (and we should) then we have to state it like this. The original form will follow `by simp` thanks to the other suggestions.